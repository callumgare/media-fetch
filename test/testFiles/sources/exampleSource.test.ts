import { expect } from "vitest";
import { createBasicTestsForRequestHandlers } from "../../utils/vitest.js";
import exampleSource from "../../fixtures/exampleSource.js";

createBasicTestsForRequestHandlers({
  source: exampleSource,
  queries: {
    "single-media": {
      request: { id: "test-getWebpage" },
      checkResponse: (response) => expect(response).toMatchSnapshot(),
    },
    "search-media": {
      request: { searchText: "search text" },
      checkResponse: (response) => expect(response).toMatchSnapshot(),
      numOfPagesToLoad: 3,
      numOfPagesToExpect: 2,
    },
  },
  queriesShared: {
    finderOptions: {
      plugins: [
        {
          sources: [exampleSource],
        },
      ],
    },
  },
});
