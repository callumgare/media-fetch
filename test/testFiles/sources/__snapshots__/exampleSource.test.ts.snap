// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Run query "Search media" with: { "searchText": "search text" } 1`] = `
{
  "media": [
    {
      "files": [],
      "id": "1-a",
      "mediaFinderSource": "example-source",
      "title": "Media Title",
    },
  ],
  "page": {
    "isLastPage": false,
    "pageFetchLimitReached": false,
    "pageNumber": 1,
    "paginationType": "offset",
  },
  "request": {
    "pageNumber": 1,
    "queryType": "search-media",
    "searchText": "search text",
    "source": "example-source",
  },
}
`;

exports[`Run query "Search media" with: { "searchText": "search text" } 2`] = `
{
  "media": [
    {
      "files": [],
      "id": "2-a",
      "mediaFinderSource": "example-source",
      "title": "Media Title",
    },
  ],
  "page": {
    "isLastPage": true,
    "pageFetchLimitReached": false,
    "pageNumber": 2,
    "paginationType": "offset",
  },
  "request": {
    "pageNumber": 2,
    "queryType": "search-media",
    "searchText": "search text",
    "source": "example-source",
  },
}
`;

exports[`Run query "Search media" with: { "searchText": "search text" } 3`] = `null`;

exports[`Run query "Single media" with: { "id": "test-getWebpage" } 1`] = `
{
  "media": [
    {
      "files": [],
      "id": "1234",
      "mediaFinderSource": "example-source",
      "title": "Example Domain",
    },
  ],
  "request": {
    "id": "test-getWebpage",
    "queryType": "single-media",
    "source": "example-source",
  },
}
`;
